include/internal/quic_stream_map.h

```diff
@@ -503,6 +503,40 @@ static ossl_inline ossl_unused int ossl_quic_stream_recv_get_final_size(const QU
     }
 }
 
+/*
+ * Determines the number of bytes available still to be read, and whether a FIN
+ * has yet to be read.
+ */
+static ossl_inline ossl_unused int ossl_quic_stream_recv_pending(const QUIC_STREAM *s)
+{
+    size_t avail;
+    int fin = 0;
+
+    switch (s->recv_state) {
+    default:
+    case QUIC_RSTREAM_STATE_NONE:
+        return 0;
+
+    case QUIC_RSTREAM_STATE_RECV:
+    case QUIC_RSTREAM_STATE_SIZE_KNOWN:
+    case QUIC_RSTREAM_STATE_DATA_RECVD:
+        if (!ossl_quic_rstream_available(s->rstream, &avail, &fin))
+            avail = 0;
+
+        if (avail == 0 && fin)
```

Comment:
question/comment: I think it should actually count `fin` as byte here:
```
if (fin)
    avail++;
```

Reply:
This only applies if avail is 0, so the outcome is the same either way...

Reply:
The reason I was asking was we could potentially save one call to `ossl_quic_stream_recv_pending()` (I think). In case the fin as already received with last chunk of data (X+1).

It's a detail anyway.